### DISCUSSION TITLE
May I ask how I want to create my own thread when using C to extend MicroPython?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12612] [ESP32]
I wanted to use EventLoop in MicroPython, but the problem was that when using m_new_obj to create objects in the event_handler function, or an error caused the system to crash. At this point, I tried to add mp_thread_init at the beginning of the event_handler function, and the crash problem was resolved. However, when using the mp_call_function_1 callback function, an error was reported.

I guess this is a problem caused by MicroPython's multithreading mechanism. What should I do when I want to create my own thread when using C to extend MicroPython?

**Comment by jimmo** [comment]
See also https://github.com/orgs/micropython/discussions/12596#discussioncomment-7226011
